drivers/gpio/gpio_grgpio2.c

```diff
@@ -0,0 +1,280 @@
+/*
+ * Copyright (c) 2021 Cobham Gaisler AB
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * Driver for GRLIB GRGPIO revision 2.
+ * - iflag determine pending interrupt.
+ * - interrupt map decides interrupt numer if implemented.
+ * - logic or/and/xor registers used when possible
+ */
+
+#define DT_DRV_COMPAT gaisler_grgpio
+
+#include <drivers/gpio.h>
+#include <spinlock.h>
+#include "gpio_utils.h"
+#include "gpio_grgpio.h"
+
+#include <logging/log.h>
+#define LOG_LEVEL CONFIG_GPIO_LOG_LEVEL
+LOG_MODULE_REGISTER(gpio_grgpio2);
+
+struct cfg {
+	volatile struct grgpio_regs *regs;
+	int interrupt;
+};
+
+struct data {
+	struct gpio_driver_data common;
+	struct k_spinlock lock;
+	sys_slist_t cb;
+	uint32_t imask;
+	uint32_t connected;
+	int irqgen;
+};
+
+static void grgpio_isr(const struct device *dev)
+{
+	const struct cfg *cfg = dev->config;
+	struct data *data = dev->data;
+	volatile struct grgpio_regs *regs = cfg->regs;
+	uint32_t pins;
+
+	/* no locking needed when iflag is implemented */
+	pins = regs->iflag;
+	if (pins == 0) {
+		return;
+	}
+	regs->iflag = pins;
+	gpio_fire_callbacks(&data->cb, dev, pins);
+}
+
+static int init(const struct device *dev)
```

Comment:
It still happens that people need to read disassembly code. A short, popular names like `init` make it difficult. Let's prefix the function, e.g. `grgpio2_init`.

Reply:
Done